doc/Customization-guide/How-to-contribute/Git-flow/index.md

```diff
@@ -0,0 +1,101 @@
+
+```sh
+git clone git@github.com:your_github_name/kaa.git
+cd kaa
+```
+
+You may need to sync with the main repo, so it's good to add it to remotes:
+
+```sh
+git remote add upstream https://github.com/kaaproject/kaa.git
+```
+
+The rest of the guide assumes upstream points to `kaaproject/kaa`.
+
+## Merge requirements
+
+### Branch off from appropriate branch
+
+First of all, you should decide what branch you will base your changes on. Generally, that's the `develop` branch.
+
+## Merge requirements
+
+### Branch off from appropriate branch
+
+First of all, you should decide what branch you will base your changes on. Generally, that's the `develop` branch.
+
+The rules are next:
+
+- `release-xxx` if your change is a **bugfix** that should go to the release *xxx* and there is such branch. That's either updates to patch releases or pre-release hotfixes.
+- `develop` if you want your change in the next minor version bump. (Minor is a second version component. e.g. 0.8.1 -> 0.9.0 is a minor bump. Note also that 0.9.1 -> 1.0.0 is a minor bump as well.)
+- `develop-xxx` if your change shouldn't go to the next release. Usually, you know that; if you don't, that's a wrong branch.
+- `master` -- never.
+
+Take into account fix version field in Jira.
+
+Note also that that's is a branch you will open pull request against later.
+
+It's always a good idea to fetch newest changes from the branch before starting:
+
+```sh
+git checkout develop # assuming you branch off from develop
+git pull
+```
+
+### Opened against appropriate branch
+
+You must open the pull request against the branch you've branched from. That means no pull requests to master are accepted.
+
+### No merge conflicts
+
+The pull request shouldn't have have any merge conflicts. If there is one, you should resolve it and update pull request.
+
+You can do that with the following commands:
+
+```sh
+# checkout a branch you open PR from
+git fetch upstream # assuming upstream is kaaproject/kaa
+git merge upstream/merge_branch # Where merge_branch is a branch you open pull request against.
+# resolve pull requests
+git add changed_files
+git commit
+git push
+```
+
+GitHub will automatically update your pull request.
+
+### Tests
+
+All pull requests are automatically checked with Travis and Jenkins. If build fails any test, the pull request must not be merged.
+
+### Review
+
+The pull request must have at least two LGTM (Looks Good To Me) from the members of the team responsible for a changed component and shouldn't have any open review questions.
+
+## Gatekeepers
+
+Gatekeeper is a person who is responsible for the final review and merge; he is also responsible for managing git repos. That is the only person who can write to master, develop and release branches.
+
+If a pull request satisfies all merge requirements, one of gatekeepers must do a final review and merge.
```

Comment:
"one of gatekeepers must do a final review and merge."
Could this be changed to 
"one of gatekeepers should do a final review and merge."